cqf-fhir-utility/src/main/java/org/opencds/cqf/fhir/utility/adapter/dstu3/MeasureAdapter.java

```diff
@@ -44,18 +48,34 @@ public Measure copy() {
     private Library effectiveDataRequirements;
     private LibraryAdapter effectiveDataRequirementsAdapter;
 
+    private String getEdrReferenceString(Extension edrExtension) {
+        return edrExtension.getUrl().contains("cqfm")
+                ? ((Reference) edrExtension.getValue()).getReference()
+                : ((UriType) edrExtension.getValue()).getValue();
+    }
+
+    private Consumer<String> getEdrReferenceConsumer(Extension edrExtension) {
+        return edrExtension.getUrl().contains("cqfm")
+                ? (reference) -> edrExtension.setValue(new Reference(reference))
+                : (reference) -> edrExtension.setValue(new UriType(reference));
+    }
+
     private void findEffectiveDataRequirements() {
         if (!checkedEffectiveDataRequirements) {
-            var edrExtensions = this.getMeasure().getExtension().stream()
+            List<Extension> edrExtensions = this.getMeasure().getExtension().stream()
                     .filter(ext -> ext.getUrl().endsWith("-effectiveDataRequirements"))
                     .filter(ext -> ext.hasValue())
                     .collect(Collectors.toList());
 
             var edrExtension = edrExtensions.size() == 1 ? edrExtensions.get(0) : null;
-            if (edrExtension != null) {
-                var edrReference = ((Reference) edrExtension.getValue()).getReference();
+            // cqfm-effectiveDataRequirements is a Reference, crmi-effectiveDataRequirements is a canonical
+            var maybeEdrReference = Optional.ofNullable(edrExtension).map(e -> getEdrReferenceString(e));
+            if (maybeEdrReference.isPresent()) {
+                var edrReference = maybeEdrReference.get();
                 for (var c : getMeasure().getContained()) {
-                    if (c.hasId() && String.format("#%s", c.getId()).equals(edrReference) && c instanceof Library) {
+                    if (c.hasId()
+                            && (edrReference.equals(c.getId()) || edrReference.equals("#" + c.getId()))
```

Comment:
`edrReference.equals(c.getId()) || edrReference.equals("#" + c.getId())`
is because if its a Reference, then HAPI strips out the "#" but if it's a CanonicalType/UriType then HAPI doesn't strip out the "#"